faust/transport/consumer.py

```diff
@@ -1111,12 +1119,18 @@ async def on_task_error(self, exc: BaseException) -> None:
         """Call when processing a message failed."""
         await self.commit()
 
-    def _add_gap(self, tp: TP, offset_from: int, offset_to: int) -> None:
+    async def _add_gap(self, tp: TP, offset_from: int, offset_to: int) -> None:
         committed = self._committed_offset[tp]
         gap_for_tp = self._gap[tp]
-        for offset in range(offset_from, offset_to):
-            if committed is None or offset > committed:
-                gap_for_tp.append(offset)
+        if committed is not None:
```

Comment:
Why not like this - https://github.com/yoav-orca/faust/commit/5525970bb4438601120c7c8d3312663ad337bfb3#diff-ea4f78eb8b3285c18b081af5b3264671021a2b51cd5a77b36ec38ba1152e51ffR1095 since the offset is linear